File: starfightersquadronsweb/src/main/js/Phase.js
define(function()
{
    "use strict";
    var Phase =
    {
        PLANNING_START: "planningStart",
        PLANNING_END: "planningEnd",
        ACTIVATION_START: "activationStart",
        ACTIVATION_REVEAL_DIAL: "activationRevealDial",
        ACTIVATION_EXECUTE_MANEUVER: "activationExecuteManeuver",
        ACTIVATION_PERFORM_ACTION: "activationPerformAction",
        ACTIVATION_END: "activationEnd",
        COMBAT_START: "combatStart",
        COMBAT_DECLARE_TARGET: "combatDeclareTarget",
        COMBAT_ROLL_ATTACK_DICE: "combatRollAttackDice",
        COMBAT_MODIFY_ATTACK_DICE: "combatModifyAttackDice",
        COMBAT_ROLL_DEFENSE_DICE: "combatRollDefenseDice",
        COMBAT_MODIFY_DEFENSE_DICE: "combatModifyDefenseDice",
        COMBAT_DEAL_DAMAGE: "combatDealDamage",
        COMBAT_END: "combatEnd",
        END_START: "endStart",
        END_END: "endEnd",
        properties:
        {
            "planningStart":
            {
                displayName: "Planning (start)",
                value: "planningStart",
            },
            "planningEnd":
            {
                displayName: "Planning (end)",
                value: "planningEnd",
            },
            "activationStart":
            {
                displayName: "Activation (start)",
                value: "activationStart",
            },
            "activationRevealDial":
            {
                displayName: "Activation (reveal dial)",
                value: "activationRevealDial",
            },
            "activationExecuteManeuver":
            {
                displayName: "Activation (execute maneuver)",
                value: "activationExecuteManeuver",
            },
            "activationPerformAction":
            {
                displayName: "Activation (perform action)",
                value: "activationPerformAction",
            },
            "activationEnd":
            {
                displayName: "Activation (end)",
                value: "activationEnd",
            },
            "combatStart":
            {
                displayName: "Combat (start)",
                value: "combatStart",
            },
            "combatDeclareTarget":
            {
                displayName: "Combat (declare target)",
                value: "combatDeclareTarget",
            },
            "combatRollAttackDice":
            {
                displayName: "Combat (roll attack dice)",
                value: "combatRollAttackDice",
            },
            "combatModifyAttackDice":
            {
                displayName: "Combat (modify attack dice)",
                value: "combatModifyAttackDice",
            },
            "combatRollDefenseDice":
            {
                displayName: "Combat (roll defense dice)",
                value: "combatRollDefenseDice",
            },
            "combatModifyDefenseDice":
            {
                displayName: "Combat (modify defense dice)",
                value: "combatModifyDefenseDice",
            },
            "combatDealDamage":
            {
                displayName: "Combat (deal damage)",
                value: "combatDealDamage",
            },
            "combatEnd":
            {
                displayName: "Combat (end)",
                value: "combatEnd",
            },
            "endStart":
            {
                displayName: "End (start)",
                value: "endStart",
            },
            "endEnd":
            {
                displayName: "End (end)",
                value: "endEnd",
            },
        },
        values: [ "planningStart", "planningEnd", "activationStart", "activationRevealDial",
                "activationExecuteManeuver", "activationPerformAction", "activationEnd", "combatStart",
                "combatDeclareTarget", "combatRollAttackDice", "combatModifyAttackDice", "combatRollDefenseDice",
                "combatModifyDefenseDice", "combatDealDamage", "combatEnd", "endStart", "endEnd", ],
    };

    if (Object.freeze)
    {
        Object.freeze(Phase);
    }

    return Phase;
});

File: starfightersquadrons/src/main/java/org/vizzini/starfightersquadrons/Phase.java
package org.vizzini.starfightersquadrons;

/**
 * Provides an enumeration of game phases.
 */
public enum Phase
{
    /** Phase. */
    PLANNING_START("Planning (start)"),

    /** Phase. */
    PLANNING_END("Planning (end)"),

    /** Phase. */
    ACTIVATION_START("Activation (start)"),

    /** Activation phase. */
    ACTIVATION_REVEAL_DIAL("Activation (reveal dial)"),

    /** Activation phase. */
    // ACTIVATION_SET_TEMPLATE("Activation (set template)"),

    /** Activation phase. */
    ACTIVATION_EXECUTE_MANEUVER("Activation (execute maneuver)"),

    /** Activation phase. */
    // ACTIVATION_CHECK_PILOT_STRESS("Activation (check pilot stress)"),

    /** Activation phase. */
    // ACTIVATION_CLEAN_UP("Activation (clean up)"),

    /** Activation phase. */
    ACTIVATION_PERFORM_ACTION("Activation (perform action)"),

    /** Phase. */
    ACTIVATION_END("Activation (end)"),

    /** Phase. */
    COMBAT_START("Combat (start)"),

    /** Combat phase. */
    COMBAT_DECLARE_TARGET("Combat (declare target)"),

    /** Combat phase. */
    COMBAT_ROLL_ATTACK_DICE("Combat (roll attack dice)"),

    /** Combat phase. */
    COMBAT_MODIFY_ATTACK_DICE("Combat (modify attack dice)"),

    /** Combat phase. */
    COMBAT_ROLL_DEFENSE_DICE("Combat (roll defense dice)"),

    /** Combat phase. */
    COMBAT_MODIFY_DEFENSE_DICE("Combat (modify defense dice)"),

    /** Combat phase. */
    // COMBAT_COMPARE_RESULTS("Combat (compare results)"),

    /** Combat phase. */
    COMBAT_DEAL_DAMAGE("Combat (deal damage)"),

    /** Phase. */
    COMBAT_END("Combat (end)"),

    /** Phase. */
    END_START("End (start)"),

    /** End phase. */
    // END_REMOVE_EVADE_TOKENS("End (remove evade tokens)"),

    /** End phase. */
    // END_REMOVE_FOCUS_TOKENS("End (remove focus tokens)"),

    /** Phase. */
    END_END("End (end)");

    /** Display name. */
    private final String displayName;

    /**
     * Construct this object.
     * 
     * @param displayName Display name.
     */
    @SuppressWarnings("hiding")
    private Phase(final String displayName)
    {
        this.displayName = displayName;
    }

    /**
     * @return the displayName
     */
    public String getDisplayName()
    {
        return displayName;
    }

    @Override
    public String toString()
    {
        return displayName;
    }
}
